hoomd/md/methods.py

```diff
@@ -580,6 +583,144 @@ def _attach(self):
+            from the manifold in terms of the implicit function. Defaults to 1e-5
+
+    :py:class:`NVE_Rattle` performs similar to :py:class:`NVE` constant volume, 
+    constant energy simulations using the standard Velocity-Verlet method. 
+    In addition particles are constrained to a manifold surface using the
+    RATTLE algorithm. For poor initial conditions that include overlapping atoms, 
+    a limit can be specified to the movement a particle is allowed to make in 
+    one time step. After a few thousand time steps with the limit set, the 
+    system should be in a safe state to continue with unconstrained integration.
+
+    Warning:
+        The particles have to be initialised close to the implicit surface of
+        the manifold.
+
+    For the MTK equations of motion, see:
+
+    * `S. Paquay and R. Kusters  2016 <https://doi.org/10.1016/j.bpj.2016.02.017>`_
+
+    .. todo::
+        Update when zero momentum updater is added.
+
+    Examples::
+        
+        sphere = hoomd.md.manifold.Sphere(r=10)
+        nve_rattle = hoomd.md.methods.NVE_Rattle(filter=hoomd.filter.All(),maifold=sphere)
+        integrator = hoomd.md.Integrator(dt=0.005, methods=[nve_rattle], forces=[lj])
+
+
+    Attributes:
+        filter (hoomd.filter.ParticleFilter): Subset of particles on which to
+            apply this method.
+
+        manifold (hoomd.md.manifold.Manifold): Manifold constraint which is 
+            used by the RATTLE algorithm of this method.
+
+        limit (None or float): Enforce that no particle moves more than a
+            distance of a limit in a single time step. Defaults to None
+
+        eta (float): Defines the error particles are allowed to deviate
+            from the manifold in terms of the implicit function. Defaults to 1e-5
+
+    """
+
+    def __init__(self, filter, manifold, limit=None, eta=0.000001):
+
+        self._manifold = validate_manifold(manifold)
+        # store metadata
+        param_dict = ParameterDict(
+            filter=ParticleFilter,
+            limit=OnlyType(float, allow_none=True),
+            zero_force=OnlyType(bool, allow_none=False),
+            eta=float(eta)
+        )
+        param_dict.update(dict(filter=filter, limit=limit, zero_force=False))
+
+        # set defaults
+        self._param_dict.update(param_dict)
+
+    def _attach(self):
+        if not self._manifold._added:
+            self._manifold._add(self._simulation)
+        else:
+            if self._simulation != self._manifold._simulation:
+                raise RuntimeError("{} object's manifold is used in a "
+                                   "different simulation.".format(type(self)))
+        if not self.manifold._attached:
+            self.manifold._attach()
+
+        # initialize the reflected c++ class
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            if self.manifold.name == "Cylinder":
+                my_class = _md.TwoStepRATTLENVECylinder
```

Comment:
There is a way to generate class names from computed strings. That would simplify this code.
`getattr(_md, 'TwoStepRATTLENVE' + self.manifold.name)`.

Reply:
Also need to programatically add `... + 'GPU' if isinstance(self._simulation.device, hoomd.device.GPU)`